qualtran/bloqs/rotations/phase_gradient.py

```diff
@@ -0,0 +1,122 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from functools import cached_property
+
+import attrs
+import cirq
+from numpy.typing import NDArray
+
+from qualtran import GateWithRegisters, Register, Side, Signature
+from qualtran.bloqs.basic_gates import Hadamard
+from qualtran.bloqs.basic_gates.rotation import CZPowGate, ZPowGate
+from qualtran.bloqs.on_each import OnEach
+
+
+@attrs.frozen
+class PhaseGradientSchoolBook(GateWithRegisters):
+    r"""A naive implementation (Controlled-/)PhaseGradient gate on an n-bit register.
+
+    Supports both
+    $$
+        \text{PhaseGrad}_{n} = \sum_{k=0}^{2^{n}-1}|k\rangle\ langle k| \omega_{n, t}^{k}
+    $$
+    and
+    $$
+        \text{CPhaseGrad}_{n} = \sum_{k=0}^{2^{n}-1} |0\rangle \langle 0| I + |1\rangle \langle1| \omega_{n, t}^{k}$
```

Comment:
too many dollar

Reply:
I don't think we need this. Just say "or a controlled version thereof". Otherwise you'll have to fire up a jupyter notebook to render the latex to see what it actually looks like to see what's different and it's just the normal formula for "controlled"